README.md

```diff
@@ -135,6 +222,66 @@ const game = Game.build({is_over: true});
 
 A tool like [babel](https://babeljs.io) is currently required to use this syntax.
 
+## Rosie API
+
+As stated above the rosie factory signatures can be broken into factory definition and object creation.
+
+Additionally factories can be defined and accessed via the Factory singleton, or they can be created and maintained by the callee.
+
+###Factory declaration functions
+
+Once you have an instance returned from a `Factory.define` or a `new Factory()` call, you do the actual of work of defining the objects. This is done using the methods below (note these are typically chained together as in the examples above):
+
+#### Factory.define
+
+* **Factory.define(``factory_name``)** - Defines a factory by name. Return an instance of a Factory that you call `.attr`, `.option`, `.sequence`, and `.after` on the result to define the properties of this factory.
+* **Factory.define(`factory_name`, `constructor`)** - Optionally pass a constuctor function, and the objects produced by `.build` will be passed through the `constructor` funtion.
+
+
+#### instance.attr:
+
+Use this to define attributes of your objects
+
+* **instance.attr(`attribute_name`, `default_value`)** - `attribute_name` is required and is a string, `default_value` is the value to use by default for the attribute
+* **instance.attr(`attribute_name`, `generator_function`)** - `generator_function` is called to generate the value of the attribute
+* **instance.attr(`attribute_name`, `dependencies`, `generator_function`)** - `dependencies` is an array of strings, each string is the name of an attribute or option that is required by the `generator_function` to generate the value of the attribute. This string of `dependencies` will match the parameters that are passed to the `generator_function`
+
+#### instance.option:
+
+Use this to define options used to programatically generate attributes of your objects
+
+* **instance.option(option_name, `default_value`)** - option_name is required and is a string, `default_value` is the value to use by default for the option
+* **instance.option(option_name, `generator_function`)** - `generator_function` is called to generate the value of the option
+* **instance.option(option_name, `dependencies`, `generator_function`)** - `dependencies` is an array of strings, each string is the name of an attribute or option that is required by the `generator_function` to generate the value of the option. This string of `dependencies` will match the parameters that are passed to the `generator_function`
+
+#### instance.sequence:
+
+Use this to define an auto incrementing sequence field in your object
+
+* **instance.sequence(`sequence_name`)** - define a sequence called `sequence_name`, set the start value to 1
+* **instance.sequence(`sequence_name`, `generator_function`)** - `generator_function` is called to generate the value of the sequence
```

Comment:
This should indicate that the generator function may take the sequence numbers as an argument.